db.sh

```diff
@@ -2,51 +2,110 @@
 
 # Get value by key
 get () {
-   db=$1;
-   key=$(echo -n "$2" | base64 -w 0);
-   sed -nr "s/^$key\ (.*$)/\1/p" $db | base64 -d
+   if [[ -z "$1" ]];
+   then
+      echo "No database provided"
+   elif [[ "$1" == "--help" ]];
+   then
+      echo "Usage: db get <database> <key>"
+   elif [[ -z "$2" ]];
+   then
+      echo "No key provided"
+   else
+      db=$1;
+      key=$(echo -n "$2" | base64 -w 0);
+      sed -nr "s/^$key\ (.*$)/\1/p" $db | base64 -d
+   fi
 }
 
 # List all keys
 list () {
-   db=$1;
-   sed -nr "s/(^[^\ ]+)\ (.*$)/\1/p" $db | xargs -n 1 sh -c 'echo `echo -n $0 | base64 -d`'
+   if [[ -z "$1" ]];
+   then
+      echo "No database provided"
+   elif [[ "$1" == "--help" ]];
+   then
+      echo "Usage: db list <database>"
+   else
+      db=$1;
+      sed -nr "s/(^[^\ ]+)\ (.*$)/\1/p" $db | xargs -n 1 sh -c 'echo `echo -n $0 | base64 -d`'
+   fi
 }
 
 # Get last added value 
 last () {
-    db=$1;
-    sed -nr "\$s/(.*)\ (.*$)/\2/p;d" $db | base64 -d
+   if [[ -z "$1" ]];
+   then
+      echo "No database provided"
+   elif [[ "$1" == "--help" ]];
+   then
+      echo "Usage: db last <database>"
+   else
+      db=$1;
+      sed -nr "\$s/(.*)\ (.*$)/\2/p;d" $db | base64 -d
+   fi
 }
 
 # Put or updated record
 put () {
-    db=$1;
-    key=$(echo -n "$2" | base64 -w 0);
-    value=$3;
-    if [ -z "$3" ]
-    then
-       value=$(base64 -w 0 <&0);
-    else
-       value=$(echo -n "$value" | base64 -w 0);
-    fi
+   if [[ -z "$1" ]];
+   then
+      echo "No database provided"
+   elif [[ "$1" == "--help" ]];
+   then
+      echo "Usage: db put <database> <key> ?<value>"
+   elif [[ -z "$2" ]];
+   then
+      echo "No key provided"
+   else
+      db=$1;
+      key=$(echo -n "$2" | base64 -w 0);
+      value=$3;
+      if [ -z "$3" ]
+      then
+         value=$(base64 -w 0 <&0);
+      else
+         value=$(echo -n "$value" | base64 -w 0);
+      fi
 
-    if [ ! -f "$1" ]; then touch "$db"; fi;
-    if [[ $(grep "^"$key"\ " $db) == "" ]]; then
-        #Insert
-        echo "$key $value" >> $db
-    else
-        #Replace
-        sed -ir "s/^$key\ .*/$key $value/g" $db
-    fi;
+      if [ ! -f "$1" ]; then touch "$db"; fi;
+      if [[ $(grep "^"$key"\ " $db) == "" ]]; then
+         #Insert
+         echo "$key $value" >> $db
+      else
+         #Replace
+         sed -ir "s/^$key\ .*/$key $value/g" $db
+      fi;
+   fi;
 }
 
 # Remove record by key
 delete () {
-   db=$1;
-   key=$(echo -n "$2" | base64 -w 0);
-   sed -ri "/^"$key"\ .*/d" "$db"
+   if [[ -z "$1" ]];
+   then
+      echo "No database provided"
+   elif [[ "$1" == "--help" ]];
+   then
+      echo "Usage: db delete <database> <key>"
+   elif [[ -z "$2" ]];
+   then
+      echo "No key provided"
```

Comment:
(here, above, and bellow) Seems like 

```
echo "..."
exit 1
```

will remove the following `else` and allows user to rely on the exit code

Reply:
That looks reasonable.
I will definitely work on that.